sunnypilot/selfdrive/controls/lib/accel_personality/accel_controller.py

```diff
@@ -0,0 +1,91 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# Last updated: Feb 1, 2025
+from cereal import custom
+from openpilot.common.numpy_fast import interp
+from openpilot.common.realtime import DT_MDL
+from openpilot.common.params import Params
+
+AccelPersonality = custom.LongitudinalPlanSP.AccelerationPersonality
+
+# Accel personality by @arne182 modified by cgw and kumar
+
+_DP_CRUISE_MIN_V_ECO =    [-0.02, -0.02, -0.85, -0.85, -0.85, -0.85, -1.2,  -1.2]
+_DP_CRUISE_MIN_V_NORMAL = [-0.02, -0.02, -0.90, -0.90, -0.90, -0.90, -1.2,  -1.2]
+_DP_CRUISE_MIN_V_SPORT =  [-0.02, -0.02, -1.0,  -1.0,  -1.0,  -1.0,  -1.2,  -1.2]
+_DP_CRUISE_MIN_BP =       [0.,     2.0,    2.01,   15.,   15.01, 20.,   20.01, 40.]
+
+_DP_CRUISE_MAX_V_ECO =    [1.60, 1.60, 1.58, 1.58, 0.86, .532, .432, .32,  .28,  .085]
+_DP_CRUISE_MAX_V_NORMAL = [1.80, 1.80, 1.75, 1.75, 0.92, .65,  .56,  .36,  .30,  .12]
+_DP_CRUISE_MAX_V_SPORT =  [2.00, 2.00, 2.00, 2.00, 1.05, .79,  .61,  .42,  .333, .2]
+_DP_CRUISE_MAX_BP =       [0.,   1.,   6.,   8.,   11.,  16,   20.,  25.,  30.,  55.]
+
+
+class AccelController:
+  def __init__(self):
+    self._params = Params()
+    self._personality = AccelPersonality.stock
+    self._frame = 0
+
+  def _read_params(self):
+    if self._frame % int(1. / DT_MDL) == 0:
+      personality_str = self._params.get("AccelPersonality", encoding='utf-8')
+      if personality_str is not None:
+        personality_int = int(personality_str)
+        if personality_int in [AccelPersonality.stock, AccelPersonality.normal, AccelPersonality.eco, AccelPersonality.sport]:
+          self._personality = personality_int
+
+  def _dp_calc_cruise_accel_limits(self, v_ego: float) -> tuple[float, float]:
+    self._read_params()  # Ensure personality updates
+
+    if self._personality == AccelPersonality.eco:
+      min_v = _DP_CRUISE_MIN_V_ECO
+      max_v = _DP_CRUISE_MAX_V_ECO
+      #print("ECO")
+    elif self._personality == AccelPersonality.sport:
+      min_v = _DP_CRUISE_MIN_V_SPORT
+      max_v = _DP_CRUISE_MAX_V_SPORT
+      #print("SPORT")
+    else:
+      min_v = _DP_CRUISE_MIN_V_NORMAL
+      max_v = _DP_CRUISE_MAX_V_NORMAL
+      #print("NORMAL")
```

Comment:
For the prints, we can utilize `cloudlog.debug` from [`swaglog.py`](https://github.com/sunnypilot/sunnypilot/blob/f4f48ee55c6b526b4665b2a26b42442aeb9a6cf8/common/swaglog.py) if needed.